src/lib.rs

```diff
@@ -244,7 +265,10 @@ pub struct Citation {
     pub journal: Option<String>,
     /// Journal abbreviation
     pub journal_abbr: Option<String>,
+    /// Publication date with day, month, and year
+    pub date: Date,
```

Comment:
As per https://github.com/AliAzlanDev/biblib/pull/4#issuecomment-2960175874 I think the field should be `pub date: Option<Date>` and `Date::year` should be non-optional.

It makes more sense to say "a Citation may or may not have a date" (i.e. `Option<Date>`) instead of "a Citation has a date with no year, no month, no day."

I.e.

```rust
// IMO this is preferred
Citation { date: None, .. }
Citation { date: Some(Date {year: 2025, month: None, day: None}), .. }

// This is what you have, which makes less sense:
Citation { date: Date { year: None, month: None, day: None }, .. }
Citation { date: Date { year: Some(2025), month: None, day: None }, .. }
```